### PR TITLE
Closes #601

### LINKED ISSUE #601: Do not allow to yield too long tuples
# Rule request

## Thesis

Expressions like `yield foo, bar, baz, fling, flung` should not be allowed.

## Reasoning

We should use lists of similar type or wrapper objects instead.
Related: https://github.com/troessner/reek/blob/master/docs/Long-Yield-List.md
Original: https://github.com/wemake-services/wemake-python-styleguide/issues/422

### PR DESCRIPTION
**Pull request by fullset**
# I have made things!

## Checklist

- [x] I have double checked that there are no unrelated changes in this pull request (old patches, accidental config files, etc)
- [x] I have created at least one test case for the changes I have made
- [x] I have updated the documentation for the changes I have made
- [x] I have added my changes to the `CHANGELOG.md`

## Related issues

Hi.
@sobolevn Look, please!

### CODE REVIEW

File: tests/test_visitors/test_ast/test_complexity/test_counts/test_yield_length.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    i = 0
+    while True:
+        yield i + 1, i + 2, i + 3
+"""
+
+long_yield = """
+def function_name(foo, bar, baz):
+    i = 0
+    while True:
+        yield i + 1, i + 2, i + 3, foo + 1, bar + 1, baz + 1
```
Comment (sobolevn) [inline tests/test_visitors/test_ast/test_complexity/test_counts/test_yield_length.py]: Let's do a template instead! And insert values into it.
Because `short_yield` and `long_yield` are already equal.

We should also test: `()` (empty tuple), `(1, 2, 3, 4, 5, 6)` (tuple with `(` and `)`)

File: tests/test_visitors/test_ast/test_complexity/test_counts/test_yield_length.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    code,
+    default_options,
+):
+    """Testing that violations are raised when reaching max value."""
+    tree = parse_ast_tree(code)
+
+    visitor = YieldTupleVisitor(default_options, tree=tree)
+    visitor.run()
+
+    assert_errors(visitor, [TooLongYieldTupleViolation])
```
Comment (sobolevn) [inline tests/test_visitors/test_ast/test_complexity/test_counts/test_yield_length.py]: `assert_error_text` is missing.

File: wemake_python_styleguide/visitors/ast/complexity/counts.py [hunk 3]
```diff
@@ -253,3 +255,36 @@ def visit_Try(self, node: ast.Try) -> None:
...
+@alias('visit_any_function', (
+    'visit_FunctionDef',
+    'visit_AsyncFunctionDef',
+))
+class YieldTupleVisitor(BaseNodeVisitor):
+    """Finds too long ``tuples`` in ``yield`` expressions."""
+
+    def _check_yield_values(self, node: AnyFunctionDef) -> None:
+        for sub_node in ast.walk(node):
+            if isinstance(sub_node, ast.Yield) and sub_node.value:
```
Comment (sobolevn) [inline wemake_python_styleguide/visitors/ast/complexity/counts.py]: We can directly visit `ast.Yield`, we don't need to visit functions.

**Comment by fullset** [comment]
No problem. I'm going to fix these, but a bit later.

File: tests/fixtures/noqa.py [hunk 4]
```diff
@@ -27,6 +27,20 @@ def __getattr__():  # noqa: Z413
...
+    # https://github.com/wemake-services/wemake-python-styleguide/issues/601
+    counter = 0
+    c1_in = 0
+    c2_in = 0
+    c1_out = 0
+    c2_out = 0
+
+    while counter < limit:
+        yield c1_in, c2_in, limit, c1_out, c2_out, counter  # noqa: Z227
+        counter += 1
```
Comment (sobolevn) [inline tests/fixtures/noqa.py]: Do we need this line?

File: tests/fixtures/noqa.py [hunk 5]
```diff
@@ -27,6 +27,20 @@ def __getattr__():  # noqa: Z413
...
+    # See:
+    # https://github.com/wemake-services/wemake-python-styleguide/issues/601
+    counter = 0
+    c1_in = 0
+    c2_in = 0
+    c1_out = 0
+    c2_out = 0
+
+    while counter < limit:
+        yield c1_in, c2_in, limit, c1_out, c2_out, counter  # noqa: Z227
```
Comment (sobolevn) [inline tests/fixtures/noqa.py]: Do we need all these variables? Can we `yield` just number? 
It would be shorter but have the same effect.

**Comment by fullset** [comment]
@sobolevn done.

File: tests/fixtures/noqa.py (hunk 5 above)
Comment (fullset) [reply tests/fixtures/noqa.py]: Ok. The simpler the better.

**Comment by sobolevn** [comment]
Awesome job! Thank you.
